### ISSUE TITLE
Opt-in for social media posting

### ISSUE THREAD

**Issue by jacksongoode** [pollinations/pollinations#54]
As of now I see that the results are posted to various accounts once the notebook is complete. I have two suggestions regarding this.
- At minimum, the user should be aware that posts are being made (there could be sensitive information used as prompts for generation)
- This functionality should be opt-in with a toggle next to "Submit"

**Comment by voodoohop** [comment]
Hey. This is one of first things on our todo list. There is a small sentence about it when you click the help button but its too hidden. We will add a checkbox to enable/disable these days.

At least none of the data can be associated with an individual since we are not obtaining, logging or saving any user information

**Comment by jacksongoode** [comment]
Yes, I saw that after I made the post actually. But there may be cases where someone might enter some sensitive data within the prompt, without the knowledge that will be made public, that is then sent and logged to any of these social media accounts. I really think this info ought to be made available more visibly. Thanks!

**Comment by voodoohop** [comment]
Absolutely. Will add it later today or tomorrow.

**Comment by voodoohop** [comment]
Will be part of moving the social media posting trigger to the front-end #30

**Comment by gokaykucuk** [comment]
@jacksongoode @voodoohop  what do you think about making this opt-out but very obvious? Showing a confirmation dialog for the first-time users etc. Because if it's opt-in, I'm guessing people who wouldn't mind us sharing the results might not opt-in as it requires more effort to do so and social media reach is one of the important aspects of the project.

**Comment by jacksongoode** [comment]
> @jacksongoode @voodoohop what do you think about making this opt-out but very obvious? Showing a confirmation dialog for the first-time users etc. Because if it's opt-in, I'm guessing people who wouldn't mind us sharing the results might not opt-in as it requires more effort to do so and social media reach is one of the important aspects of the project.

If it is very clear - directly next to a submit button - I can see it being alright as opt-out. I also think a first-time dialogue is a good idea as well. But I believe this direction would also require you to make certain that the content that is posted by the AI doesn't violate the standards at each of the social media platforms (filter by word/sentiment).

It would also be nice if this opt-out persisted through session via cookie.

**Comment by gokaykucuk** [comment]
@jacksongoode currently we have implemented a basic mature word filtering, but before that, we had some issues with social media platforms that's a good point. It can use some improvements but I'll park it until we have another issue from one of the platforms.

About persistence, yes I totally agree. I think implementation wise localStorage might be easier for us, wdyt @voodoohop ?

**Comment by voodoohop** [comment]
Maybe a small checkbox that is ticked by default close to the post button whose state we persist to local storage?

At the same time, it's something that once the user has made the decision whether to post to social media or not he doesn't really need to be presented with a checkbox each time he uses pollinations. Maybe in the future, we could have some unobtrusive "settings" somewhere.

I like the dialog idea but what if the user wants to change it later? having both a dialog and a checkbox seems a bit overkill at the moment. 

I'm just in the process of moving the social media posting trigger to the frontend on this branch: https://github.com/pollinations/pollinations/tree/refactor/social-media-post-from-frontend

**Comment by gokaykucuk** [comment]
@voodoohop I think the dialog idea works only for the first time the user is on pollinations.ai. Only thing the dialog will achieve is making sure the user cannot miss the checkbox next to submit button.

The subsequent visits shouldn't have the dialog, but just the checkbox. The checkbox state should be stored in localStorage so the user can update it later on.

**Comment by voodoohop** [comment]
Ok since we need the checkbox anyway I will implement that first and then we can add a dialog

**Comment by jacksongoode** [comment]
> Ok since we need the checkbox anyway I will implement that first and then we can add a dialog

Sounds like a good practice to keep some kind of small visible notice that the resulting content will be posted somewhere.

**Comment by voodoohop** [comment]
closed by #60 

<img width="1081" alt="Screenshot 2021-08-19 at 19 13 30" src="https://user-images.githubusercontent.com/5099901/130113633-a6a353c6-a432-4c72-86fc-6f54c8f44c30.png">
